public/pages/workflow_detail/workflow_detail.test.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import {
+  RouteComponentProps,
+  Route,
+  Switch,
+  Router,
+  Redirect,
+} from 'react-router-dom';
+import { WorkflowDetail } from './workflow_detail';
+import { WorkflowDetailRouterProps } from '../../pages';
+import '@testing-library/jest-dom';
+import { mockStore, resizeObserverMock } from '../../../test/utils';
+import { createMemoryHistory } from 'history';
+import { WORKFLOW_TYPE } from '../../../common';
+
+jest.mock('../../services', () => {
+  const { mockCoreServices } = require('../../../test');
+  return {
+    ...jest.requireActual('../../services'),
+    ...mockCoreServices,
+  };
+});
+
+const history = createMemoryHistory();
+window.ResizeObserver = resizeObserverMock;
+
+const renderWithRouter = (
+  workflowId: string,
+  workflowName: string,
+  workflowType: WORKFLOW_TYPE
+) => ({
+  ...render(
+    <Provider store={mockStore(workflowId, workflowName, workflowType)}>
+      <Router history={history}>
+        <Switch>
+          <Route
+            path="/workflow/:workflowId"
+            render={(props: RouteComponentProps<WorkflowDetailRouterProps>) => {
+              return <WorkflowDetail setActionMenu={jest.fn()} {...props} />;
+            }}
+          />
+          <Redirect from="/" to={`/workflow/${workflowId}`} />
+        </Switch>
+      </Router>
+    </Provider>
+  ),
+});
+
+const workflowId = '12345';
+const workflowName = 'test_workflow';
+
+describe('WorkflowDetail', () => {
+  const workflowTypes = [
+    WORKFLOW_TYPE.CUSTOM,
+    WORKFLOW_TYPE.SEMANTIC_SEARCH,
+    WORKFLOW_TYPE.HYBRID_SEARCH,
+  ];
+
+  workflowTypes.forEach((type) => {
+    test(`renders the page with ${type} type`, () => {
```

Comment:
Rather than re-defining the list of types here, you can just use the enum directly.
something like:
```
Object.values(WORKFLOW_TYPE).forEach((type) => ...)
```
I like the idea of automatically generating the test cases for each workflow type, nice!